src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,70 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private final int[] array;
+    private int left;
+    private int right;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length - 1);
+    }
+
+    private MergeSortAction(int[] array, int left, int right) {
+        this.array = array;
+        this.left = left;
+        this.right = right;
+    }
+    // 3 9 1 0 2 8
+    // mid = 2
+    // 3 9 1
+    // 3 9
+    // 1
+    // 3
+    // 9
+
+    // 0 2 8 ->
+    @Override
+    protected void compute() {
+        if (left < right) {
+            int mid = (left + right) / 2;
+            MergeSortAction leftTask = new MergeSortAction(array, left, mid);
+            MergeSortAction rightTask = new MergeSortAction(array, mid + 1, right);
+            invokeAll(leftTask, rightTask);
```

Comment:
Before invoking the subtasks, it would be beneficial to check if the current task should continue to be split. If the size of the subarray is below a certain threshold, it might be more efficient to sort it using a non-recursive method directly, to avoid the overhead of creating more RecursiveAction instances.